### PR TITLE
feat: Adding state disable to dropdown

### PR DESCRIPTION
**Pull request by palbizu**
## Description
Adding disabling state from form.

### Testing
Tested locally, visually

image

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/components/src/select/select.component.ts [hunk 1]
```diff
@@ -324,6 +324,10 @@ export class SelectComponent<V> implements ControlValueAccessor, AfterContentIni
     this.propagateControlValueChangeOnTouch = onTouch;
   }
 
+  public setDisabledState(isDisabled?: boolean): void {
```
Comment (jake-bassett) [inline projects/components/src/select/select.component.ts:327]: Since the property is public is there a need for a method to access it?

Comment (palbizu) [reply projects/components/src/select/select.component.ts:327]: I have no clear the reason. Probably @anandtiwary can clarify it. I followed the same structure already implemented in the input component to the Control Value Accessor. I will try to find the reason.

Comment (itssharmasandeep) [reply projects/components/src/select/select.component.ts:327]: Actually this is related to `ControlValueAccessor` interface, this function will be used by reactive form field to set the disabled state using form group itself. We will not pass any [disabled] input in the template while using the select component in the forms.

Comment (palbizu) [reply projects/components/src/select/select.component.ts:327]: So @itssharmasandeep , does it fine to keep it public?

**Comment by palbizu** [comment]
Closing this PR because It's not necessary anymore but we're going to adding it in the future.

File: projects/components/src/select/select.component.ts (hunk 1 above)
Comment (itssharmasandeep) [reply projects/components/src/select/select.component.ts:327]: Yeah it is fine.

**Review by itssharmasandeep — APPROVED** [review]

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**